base/operators.jl

```diff
@@ -328,8 +328,15 @@ eltype(::Type{Any}) = Any
 eltype(t::DataType) = eltype(supertype(t))
 eltype(x) = eltype(typeof(x))
 
-# function pipelining
+# function pipelining, composition & negation
+
 |>(x, f) = f(x)
+∘(f, g) = (x...)->f(g(x...))
+
+!(f::Function) = (x...)->!f(x...)
+^(f::Function, g::Function) = (x...)->f(x...) ^ g(x...)
+|(f::Function, g::Function) = (x...)->f(x...) || g(x...)
```

Comment:
It is consistent: `f | g` means `x -> f(x) || g(x)` and `f & g` means `f(x) && g(x)` and `f ^ g` means `x -> f(x) ^^ g(x)` except that `^^` is just spelled `^` since you always need to evaluate both arguments to an xor. I decided that the lazy version was likely more useful in general.